File: QTD2.Domain/Entities/Core/Version_Task_Question.cs
namespace QTD2.Domain.Entities.Core
{
    public class Version_Task_Question : Common.Entity
    {
        public int TaskQuestionId { get; set; }

        public int VersionTaskId { get; set; }

        public string Question { get; set; }

        public string Answer { get; set; }

        public int QuestionNumber { get; set; }

        public virtual Version_Task Version_Task { get; set; }

        public virtual Task_Question Task_Question { get; set; }

        public Version_Task_Question()
        {
        }

        public Version_Task_Question(Task_Question question)
        {
            TaskQuestionId = question.Id;
            VersionTaskId = question.TaskId;
            Question = question.Question;
            Answer = question.Answer;
            QuestionNumber = question.QuestionNumber;
        }

        public Version_Task_Question(Version_Task task, Task_Question question)
        {
            TaskQuestionId = question.Id;
            VersionTaskId = task.Id;
            Question = question.Question;
            Answer = question.Answer;
            QuestionNumber = question.QuestionNumber;
        }
    }
}

File: QTD.SPA/src/app/_DtoModels/Version_Task_Question/Version_Task_Question.ts
import { Entity } from '../Entity';
import { Task_Question } from '../Task_Question/Task_Question';
import { Version_Task } from '../Version_Task/Version_Task';

export class Version_Task_Question extends Entity {
  taskQuestionId!: any;
  versionTaskId!: any;
  question!: string;
  answer!: string;
  Version_Task!: Version_Task;
  task_Question!: Task_Question;
}
